### PR TITLE
433 googletest issue with m1 macos machine

### PR DESCRIPTION
**Pull request by xsaschako**
## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [x] There is at least one issue associated with the pull request.
* [x] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [x] New code adheres with the [coding guidelines](coding-guidelines)
* [x] Tests for new functionality has been added
* [x] A local test was succesful
* [x] There is appropriate **documentation** of your work. (use doxygen style comments)
* [x] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [x] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [x] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [x] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [x] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [x] There are appropriate **unit tests** and they pass.
* [x] The git history is clean and linearized for the merge request.
* [x] Coverage report for new code is acceptable.

### CODE REVIEW

**Comment by mknaranja** [comment]
Pleas always check "Checks by code author" @xsaschako

**Comment by xsaschako** [comment]
> Please always check "Checks by code author" @xsaschako

done and ready 4 review

**Comment by mknaranja** [comment]
partially addresse #409

File: cpp/memilio/io/json_serializer.h [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "memilio/utils/metaprogramming.h"
 #include "json/json.h"
 #include "boost/optional.hpp"
+//#include "boost/optional/optional_io.hpp"
```
Comment (mknaranja) [inline cpp/memilio/io/json_serializer.h:31]: pls remove

**Review by mknaranja — APPROVED** [review]

**Pull request merged**